src/ktopt_planning_context.cpp

```diff
@@ -252,6 +253,73 @@ void KTOptPlanningContext::addPathPositionConstraints(KinematicTrajectoryOptimiz
   }
 }
 
+void KTOptPlanningContext::addPathOrientationConstraints(KinematicTrajectoryOptimization& trajopt,
+                                                         const MultibodyPlant<double>& plant,
+                                                         Context<double>& plant_context, const double padding)
+{
+  // retrieve the motion planning request
+  const auto& req = getMotionPlanRequest();
+
+  // check for path position constraints
+  if (!req.path_constraints.orientation_constraints.empty())
+  {
+    for (const auto& orientation_constraint : req.path_constraints.orientation_constraints)
+    {
+      // NOTE: Current thesis, when you apply a Drake Orientation Constraint it
+      // expects the following
+      // Frame A: The frame in which the orientation constraint is defined
+      // Frame B: The frame to which the orientation constraint is enforced
+      // theta_bound: The angle difference between frame A's orientation and
+      //              frame B's orientation
+
+      // Extract FrameA and FrameB for orientation constraint
+      // frame A
+      const auto link_ee_name = orientation_constraint.link_name;
+      if (!plant.HasBodyNamed(link_ee_name))
+      {
+        RCLCPP_ERROR(getLogger(),
+                     "The link specified in the PositionConstraint message, %s, does not exist in the Drake "
+                     "plant.",
+                     link_ee_name.c_str());
+        continue;
+      }
+      const auto& link_ee_frame = plant.GetFrameByName(link_ee_name);
+
+      // frame B
+      const auto base_frame_name = orientation_constraint.header.frame_id;
+      if (!plant.HasBodyNamed(base_frame_name))
+      {
+        RCLCPP_ERROR(getLogger(),
+                     "The link specified in the PositionConstraint message, %s, does not exist in the Drake "
+                     "plant.",
+                     base_frame_name.c_str());
+        continue;
+      }
+      const auto& base_frame = plant.GetFrameByName(base_frame_name);
+
+      // Extract the orientation of that Frame B needs to be constrained to w.r.t Frame A
+      const auto& constrained_orientation = orientation_constraint.orientation;
+      // convert to drake's RotationMatrix
+      const auto R_BbarB = RotationMatrixd(Eigen::Quaterniond(constrained_orientation.w, constrained_orientation.x,
+                                                              constrained_orientation.y, constrained_orientation.z));
+
+      // NOTE: There are 3 tolerances given for the orientation constraint in moveit
+      // message, rake only takes in a single theta bound. For now, we take any
+      // of the tolerances as the theta bound
+      const double theta_bound = orientation_constraint.absolute_x_axis_tolerance;
+
+      // Add position constraint to each knot point of the trajectory
```

Comment:
I was wondering in the last PR whether we should have used the same number of knot points for all position/orientation constraints. What do you think?

Reply:
It is a parameter, so the user has the option of adding more/less constraints along the path.

Reply:
Yes, I meant should there be 2 separate parameters for position/orientation constraints, or just use the same parameter for all constraints?

Reply:
Ah I meant, right now even though im reading theta bound as the tolerance, I am using the one @sea-bass added as a parameter.

Reply:
Oh, also this comment should read "orientation constraint"